liblouis/compileTranslationTable.c

```diff
@@ -3086,428 +3091,442 @@ compileRule(FileInfo *nested, TranslationTableHeader **table,
 								break;
 							}
-							for (k = 0; k < ruleChars.length; k++) {
-								if (!getChar(ruleChars.chars[k], *table)) {
-									compileError(nested, "Character undefined");
-									ok = 0;
-									break;
+						else if (opcode == CTO_LenEmphPhrase) {
+							if ((*table)->emphRules[i][lenPhraseOffset] =
+											compileNumber(nested))
+								ok = 1;
+						} else if (opcode == CTO_EmphModeChars) {
+							if (getRuleCharsText(nested, &ruleChars)) {
+								widechar *emphmodechars = (*table)->emphModeChars[i - 1];
+								int len;
+								for (len = 0;
+										len < EMPHMODECHARSSIZE && emphmodechars[len];
+										len++)
+									;
+								if (len + ruleChars.length > EMPHMODECHARSSIZE) {
+									compileError(nested, "More than %d characters",
+											EMPHMODECHARSSIZE);
+								} else {
+									ok = 1;
+									for (k = 0; k < ruleChars.length; k++) {
+										if (!getChar(ruleChars.chars[k], *table)) {
+											compileError(nested,
+													"Emphasis mode character undefined");
+											ok = 0;
+											break;
+										}
+										emphmodechars[len++] = ruleChars.chars[k];
+									}
+								}
+							}
+						} else if (opcode == CTO_NoEmphChars) {
+							if (getRuleCharsText(nested, &ruleChars)) {
+								widechar *noemphchars = (*table)->noEmphChars[i - 1];
+								int len;
+								for (len = 0; len < NOEMPHCHARSSIZE && noemphchars[len];
+										len++)
+									;
+								if (len + ruleChars.length > NOEMPHCHARSSIZE) {
+									compileError(nested, "More than %d characters",
+											NOEMPHCHARSSIZE);
+								} else {
+									ok = 1;
+									for (k = 0; k < ruleChars.length; k++) {
+										if (!getChar(ruleChars.chars[k], *table)) {
+											compileError(nested, "Character undefined");
+											ok = 0;
+											break;
+										}
+										noemphchars[len++] = ruleChars.chars[k];
+									}
 								}
-								noemphchars[len++] = ruleChars.chars[k];
 							}
 						}
 					}
 					free(s);
+					return ok;
 				}
-			break;
+			return 0;
 		}
 		case CTO_LetterSign: {
 			// not passing pointer because compileBrailleIndicator may reallocate table
 			TranslationTableOffset ruleOffset = (*table)->letterSign;
-			ok = compileBrailleIndicator(nested, "letter sign", CTO_LetterRule,
-					&ruleOffset, noback, nofor, table);
-			(*table)->letterSign = ruleOffset;
-			break;
+			if (compileBrailleIndicator(nested, "letter sign", CTO_LetterRule,
+						&ruleOffset, noback, nofor, table)) {
+				(*table)->letterSign = ruleOffset;
+				return 1;
+			}
+			return 0;
 		}
 		case CTO_NoLetsignBefore:
 			if (getRuleCharsText(nested, &ruleChars)) {
 				if (((*table)->noLetsignBeforeCount + ruleChars.length) >
 						LETSIGNBEFORESIZE) {
 					compileError(nested, "More than %d characters", LETSIGNBEFORESIZE);
-					ok = 0;
-					break;
+					return 0;
 				}
-				for (k = 0; k < ruleChars.length; k++)
+				for (int k = 0; k < ruleChars.length; k++)
```

Comment:
Funny, I was going to suggest `for (int k = 0; ...` above where you define local vars directly in front of a loop, but I thought you're not doing it because M$ doesn't (didn't?) handle that part of C99. But now you're doing it anyway. Which way should we go. I'm all for it, but we might get into trouble.

Reply:
I see that you are doing C99 pretty consistently below, so I'd also do it above.

Reply:
Ah, no forget it. Above is a different story